client/pages/cart.jsx

```diff
@@ -69,12 +74,22 @@ export default class Cart extends React.Component {
         }).then(res => res.json())
           .then(msgStatus => {
             if (msgStatus.msg === 'success') {
-              window.location.hash = `order?orderId=${data.orderId}`;
+              this.setState({ loading: false });
               this.props.resetCart();
+              window.location.hash = `order?orderId=${data.orderId}`;
             } else if (msgStatus.msg === 'fail') {
               console.error('Failed to send confirmation email');
+              window.location.hash = 'error';
             }
+          })
+          .catch(err => {
+            console.error(err);
+            window.location.hash = 'error';
```

Comment:
Instead of pushing the user to a new page, it's better to keep them on the same page. Often people try refreshing their browser to deal with errors, but if they refresh the page in this case, it will always show an error no matter what.